### ISSUE TITLE
A webcounter

### ISSUE THREAD

**Issue by zootella** [beakerbrowser/unwalled.garden#13]
The first thing I did after making my first website in the 90s was add a web counter/hit counter. Checking back to see a higher number of visits made the experience sticky, and seeing larger numbers encouraged me to make my site bigger and better. It was miles before social media, but already didn't feel like talking to an empty room.

Figuring out how a webcounter might work on the distributed web is an interesting thought experiment. Following the design of unwalled.garden, here's how one might look and act:

```
000000789 visitors counted [Count me!]
__user1__, __user2__, __user3__ recently visited
```

Similar to comments, the count and user list are a result of the web crawler finding the appropriate record in the friend swarm of Dats.

**Comment by pfrazee** [comment]
Yeah I love this idea. I wonder if we could generalize it into some kind of multi-purpose signal about a site. The basic structure of this record is:

```json
{
  "type": "unwalled.garden/visit",
  "topic": "dat://some-site.com",
  "createdAt": "todayish"
}
```

But that's a super common structure that could be generalized to something like:

```json
{
  "type": "unwalled.garden/action",
  "topic": "dat://some-site.com",
  "action": "visited",
  "createdAt": "todayish"
}
```

And we could then index by `action` so that you say `listActions(url, 'visited')` to get the list of visits.

Not 100% sure that "action" is right. Could also be "event", or "signal", or something else.

**Comment by todrobbins** [comment]
I like this action idea as well, but we should probably think through the privacy concerns with sharing browsing history within the public profile? Just curious if this has been discussed elsewhere.

**Comment by pfrazee** [comment]
It's not automatic. It doesnt just publish every visit.

**Comment by zootella** [comment]
Yeah. In this sense, it's quite similar to Claps on Medium, now that I think about it.

But still a one-click way for visitors to make their visit known on a site, making the distributed web feel populated and friendly, rather than like a dusty static archive.

**Comment by pfrazee** [comment]
Yes exactly right

**Comment by todrobbins** [comment]
Cool cool! Thanks @zootella!

**Comment by danimesq** [comment]
> It's not automatic. It doesnt just publish every visit.
> it's quite similar to Claps on Medium

It is also useful for a download counter, or count views on a video.
But, users can create lots of profiles and manipulate it. Antispam is need.

**Comment by pfrazee** [comment]
You only see data from people in your network, so antispam is built in
